src/roboptim-core-plugin-knitro.cc

```diff
@@ -379,41 +256,211 @@ namespace roboptim
+            break;
+        case KTR_RC_INFEAS_CON_BOUNDS:
+            cout << "The constraint bounds have been determined to be infeasible." << endl;
+            break;
+        case KTR_RC_INFEAS_VAR_BOUNDS:
+            cout << "The variable bounds have been determined to be infeasible." << endl;
+            break;
+        case KTR_RC_UNBOUNDED:
+            cout << "Problem appears to be unbounded. Iterate is feasible and objective magnitude is greater than objrange. The objective function appears to be decreasing without bound, while satisfying the constraints. If the problem really is bounded, increase the size of the parameter objrange to avoid terminating with this message." << endl;
+            break;
+        case KTR_RC_ITER_LIMIT:
+            cout << "The iteration limit was reached before being able to satisfy the required stopping criteria. A feasible point was found. The iteration limit can be increased through the user option maxit." << endl;
+            break;
+        case KTR_RC_TIME_LIMIT:
+            cout << "The time limit was reached before being able to satisfy the required stopping criteria. A feasible point was found. The time limit can be increased through the user options maxtime_cpu and maxtime_real." << endl;
+            break;
+        case KTR_RC_FEVAL_LIMIT:
+            cout << "The function evaluation limit was reached before being able to satisfy the required stopping criteria. A feasible point was found. The function evaluation limit can be increased through the user option maxfevals." << endl;
+            break;
+        case KTR_RC_CALLBACK_ERR:
+            cout << "Callback function error. This termination value indicates that an error (i.e., negative return value) occurred in a user provided callback routine." << endl;
+            break;
+        case KTR_RC_LP_SOLVER_ERR:
+            cout << "LP solver error. This termination value indicates that an unrecoverable error occurred in the LP solver used in the active-set algorithm preventing the optimization from continuing." << endl;
+            break;
+        case KTR_RC_EVAL_ERR:
+            cout << "Evaluation error. This termination value indicates that an evaluation error occurred (e.g., divide by 0, taking the square root of a negative number), preventing the optimization from continuing." << endl;
+            break;
+        case KTR_RC_OUT_OF_MEMORY:
+            cout << "Not enough memory available to solve problem. This termination value indicates that there was not enough memory available to solve the problem." << endl;
+            break;
+        case KTR_RC_USER_TERMINATION:
+            cout << "Knitro has been terminated by the user." << endl;
+            break;
+        default:
+            cout << "Unhandled error code of type: " << errorCode << endl;
+            break;
```

Comment:
Also, `KTR_RC_USER_TERMINATION` should not be treated as an error, since the user terminates the optimization on purpose.